### DISCUSSION TITLE
Help for beginner

### DISCUSSION THREAD

**Discussion by Pemcode** [NVIDIA/cloud-native-core#88] [Q&A]
Hello everyone!

First of all, thank you for your work! :)

I would like to try testing the solution available at this address: [https://catalog.ngc.nvidia.com/orgs/nvidia/teams/aiworkflows/helm-charts/rag-app-multimodal-chatbot](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/aiworkflows/helm-charts/rag-app-multimodal-chatbot)

However, I've been stuck for three days trying to install the Cloud Native Stack locally. 
I want to test it on my laptop (16GB RAM, RTX 4070, 16-core AMD Ryzen 7), but I always get stuck at some point during the installation of Kubernetes/containerd or anything else.

Is it an obligation to be connected at a NVidia VMI ?
Do you have a quick tutorial to follow the steps in order ?
Do I need to have a "hosts" file if I'm only using my local machine (running Ubuntu 22.04)?

Thank you in advance for your help.

Wishing the entire team a great day!

**Comment by AumGupta** [comment]
Hey @Pemcode,

Your question is a bit generic, but I have tried to answer all your ques one by one.

> Is it an obligation to be connected at a NVidia VMI ?

No, it's not. You can deploy it on your local machine as long as your machine can run it.

> Do you have a quick tutorial to follow the steps in order ?

( To set up Kubernetes and the NVIDIA Cloud Native Stack locally)

   - **Step 1: Install Docker**
     - For Ubuntu 22.04, you can install Docker using the following commands:
       ```bash
       sudo apt update
       sudo apt install docker.io
       sudo systemctl enable --now docker
       sudo usermod -aG docker $USER
       newgrp docker
       ```

   - **Step 2: Install Kubernetes Tools (kubectl, kubelet, kubeadm)**
     - You can install these tools with the following commands:
       ```bash
       sudo apt-get update && sudo apt-get install -y apt-transport-https curl
       curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
       echo "deb https://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee -a /etc/apt/sources.list.d/kubernetes.list
       sudo apt-get update
       sudo apt-get install -y kubelet kubeadm kubectl
       sudo apt-mark hold kubelet kubeadm kubectl
       ```

   - **Step 3: Initialize Kubernetes Cluster**
     - Initialize the cluster using kubeadm:
       ```bash
       sudo kubeadm init --pod-network-cidr=192.168.0.0/16
       ```
     - Set up local kubeconfig:
       ```bash
       mkdir -p $HOME/.kube
       sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
       sudo chown $(id -u):$(id -g) $HOME/.kube/config
       ```

   - **Step 4: Install a Pod Network Add-on (e.g., Calico)**
     - Apply Calico CNI plugin:
       ```bash
       kubectl apply -f https://docs.projectcalico.org/manifests/calico.yaml
       ```

   - **Step 5: Deploy NVIDIA GPU Operator**
     - First, enable the NVIDIA GPU operator on your cluster:
       ```bash
       helm repo add nvidia https://nvidia.github.io/gpu-operator
       helm repo update
       helm install --wait --generate-name nvidia/gpu-operator
       ```
> Do I need to have a "hosts" file if I'm only using my local machine (running Ubuntu 22.04)?

If you are only using your local machine, you typically do not need to modify the `hosts` file for Kubernetes and Docker to work.


---

Hope it helps.

**Reply by Pemcode** [reply]
Thanks a lot for your complete answer, I will try soon !
